src/UserService.Business/Commands/Certificate/EditCertificateCommand.cs

```diff
@@ -23,111 +25,113 @@
-            _httpContextAccessor = httpContextAccessor;
-            _userRepository = userRepository;
-            _certificateRepository = certificateRepository;
-            _mapper = mapper;
-            _rcImage = rcImage;
-            _logger = logger;
+          imageId = response.Message.Body;
         }
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, errorMessage);
 
-        public OperationResultResponse<bool> Execute(Guid certificateId, JsonPatchDocument<EditCertificateRequest> request)
-        {
-            List<string> errors = new List<string>();
+        errors.Add(errorMessage);
+      }
+
+      return imageId;
+    }
 
-            var senderId = _httpContextAccessor.HttpContext.GetUserId();
-            var sender = _userRepository.Get(senderId);
+    public EditCertificateCommand(
+      IAccessValidator accessValidator,
+      IHttpContextAccessor httpContextAccessor,
+      IUserRepository userRepository,
+      ICertificateRepository certificateRepository,
+      IPatchDbUserCertificateMapper mapper,
+      ICreateImageDataMapper createImageDataMapper,
+      IRequestClient<ICreateImagesRequest> rcImage,
+      ILogger<EditCertificateCommand> logger)
+    {
+      _accessValidator = accessValidator;
+      _httpContextAccessor = httpContextAccessor;
+      _userRepository = userRepository;
+      _certificateRepository = certificateRepository;
+      _mapper = mapper;
+      _createImageDataMapper = createImageDataMapper;
+      _rcImage = rcImage;
+      _logger = logger;
+    }
 
-            DbUserCertificate certificate = _certificateRepository.Get(certificateId);
+    public OperationResultResponse<bool> Execute(Guid certificateId, JsonPatchDocument<EditCertificateRequest> request)
+    {
+      List<string> errors = new List<string>();
 
-            if (!(sender.IsAdmin ||
-                  _accessValidator.HasRights(Rights.AddEditRemoveUsers))
-                  && senderId != certificate.UserId)
-            {
-                throw new ForbiddenException("Not enough rights.");
-            }
+      Guid senderId = _httpContextAccessor.HttpContext.GetUserId();
+      DbUser sender = _userRepository.Get(senderId);
 
-            var imageOperation = request.Operations.FirstOrDefault(o => o.path.EndsWith(nameof(EditCertificateRequest.Image), StringComparison.OrdinalIgnoreCase));
-            Guid? imageId = null;
-            if (imageOperation != null)
-            {
-                imageId = GetImageId(JsonConvert.DeserializeObject<AddImageRequest>(imageOperation.value?.ToString()), errors);
-            }
+      DbUserCertificate certificate = _certificateRepository.Get(certificateId);
 
-            JsonPatchDocument<DbUserCertificate> dbRequest = _mapper.Map(request, imageId);
+      if (!(_accessValidator.HasRights(Rights.AddEditRemoveUsers))
+        && senderId != certificate.UserId)
+      {
+        throw new ForbiddenException("Not enough rights.");
```

Comment:
dont throw exception

Reply:
will be done in https://github.com/LT-Students/DigitalOffice-UserService/pull/122